File: remote/src/main/java/com/kelsos/mbrc/mappers/PlaylistTrackInfoMapper.java
package com.kelsos.mbrc.mappers;

import com.kelsos.mbrc.dao.PlaylistTrackInfoDao;
import com.kelsos.mbrc.dto.playlist.PlaylistTrackInfo;
import java.util.List;
import rx.Observable;

public class PlaylistTrackInfoMapper {
  public static List<PlaylistTrackInfoDao> map(List<PlaylistTrackInfo> info) {
    return Observable.from(info).map(PlaylistTrackInfoMapper::map).toList().toBlocking().first();
  }

  public static PlaylistTrackInfoDao map(PlaylistTrackInfo object) {
    PlaylistTrackInfoDao dao = new PlaylistTrackInfoDao();
    dao.setId(object.getId());
    dao.setArtist(object.getArtist());
    dao.setTitle(object.getTitle());
    dao.setPath(object.getPath());
    dao.setDateAdded(object.getDateAdded());
    dao.setDateDeleted(object.getDateDeleted());
    dao.setDateUpdated(object.getDateUpdated());
    return dao;
  }
}

File: mbrc-lib/src/main/kotlin/com/kelsos/mbrc/mappers/PlaylistTrackInfoMapper.kt
package com.kelsos.mbrc.mappers

import com.kelsos.mbrc.dao.PlaylistTrackInfoDao
import com.kelsos.mbrc.dto.playlist.PlaylistTrackInfo
import rx.Observable
import rx.functions.Func1

object PlaylistTrackInfoMapper {
    fun map(info: List<PlaylistTrackInfo>): List<PlaylistTrackInfoDao> {
        return Observable.from(info).map<PlaylistTrackInfoDao>(Func1<PlaylistTrackInfo, PlaylistTrackInfoDao> { map(it) }).toList().toBlocking().first()
    }

    fun map(item: PlaylistTrackInfo): PlaylistTrackInfoDao {
        val dao = PlaylistTrackInfoDao()
        dao.id = item.id
        dao.artist = item.artist
        dao.title = item.title
        dao.path = item.path
        dao.dateAdded = item.dateAdded
        dao.dateDeleted = item.dateDeleted
        dao.dateUpdated = item.dateUpdated
        return dao
    }
}
